src/backend/distributed/planner/multi_explain.c

```diff
@@ -99,10 +175,728 @@ MultiExplainOneQuery(Query *query, IntoClause *into, ExplainState *es,
 		appendStringInfo(es->str, "%s\n", formattedPlanString);
 	}
 
-	/* if explain printing isn't enabled, print error only after planning */
-	if (!ExplainMultiLogicalPlan && !ExplainMultiPhysicalPlan)
+	if (!ExplainDistributedQueries)
 	{
 		appendStringInfo(es->str, "explain statements for distributed queries ");
-		appendStringInfo(es->str, "are currently unsupported\n");
+		appendStringInfo(es->str, "are not enabled\n");
+		return;
+	}
+
+	if (es->format == EXPLAIN_FORMAT_TEXT)
+	{
+		appendStringInfoSpaces(es->str, es->indent * 2);
+		appendStringInfo(es->str, "Distributed Query");
+
+		if (multiPlan->masterTableName != NULL)
+		{
+			appendStringInfo(es->str, " into %s", multiPlan->masterTableName);
+		}
+
+		appendStringInfo(es->str, "\n");
+
+		es->indent += 1;
+	}
+
+	routerExecutablePlan = RouterExecutablePlan(multiPlan, TaskExecutorType);
+
+	if (routerExecutablePlan)
+	{
+		ExplainPropertyText("Executor", "Router", es);
+	}
+	else if (TaskExecutorType == MULTI_EXECUTOR_REAL_TIME)
+	{
+		ExplainPropertyText("Executor", "Real-Time", es);
+	}
+	else if (TaskExecutorType == MULTI_EXECUTOR_TASK_TRACKER)
+	{
+		ExplainPropertyText("Executor", "Task-Tracker", es);
+	}
+	else
+	{
+		ExplainPropertyText("Executor", "Other", es);
```

Comment:
I'd use a switch, because that nicely warns if a new executor type has been added. But whatever.